TODO.md

```diff
@@ -0,0 +1,11 @@
+# Todo list
+
+/src/CapabilityService.WebApi/Infrastructure/Api/CapabilityController.cs : 109 / 131
+replace Guid.TryCatch with a value object
+give propper response code if validation fails
+
+/src/CapabilityService.WebApi/Infrastructure/Messaging/Outbox.cs
+Rename variables to reflect types used
+
+/src/CapabilityService.WebApi/Infrastructure/Persistence/CapabilityRepository.cs : 33
+Remove SaveChangesAsync and let the transaction decorator take care of the save
```

Comment:
/src/CapabilityService.WebApi/Infrastructure/Persistence/CapabilityTransactionalDecorator.cs : *
I am not sure what we are doing here is healthy. In general most of the methods only calls SaveChangesAsync once, which makes creation local transaction scopes a redundant waste of resources, also doing this makes it impossible for us to use execution strategies (see below)

/src/CapabilityService.WebApi/Infrastructure/Persistence/
We should probably look at implementing connection resilience our Db context @ https://docs.microsoft.com/en-us/ef/core/miscellaneous/connection-resiliency via execution strategies.

Reply:
I agree there is something dinky here. I will  dig a little deeper when we look into a refactoring.
`Note added todolist`